app.py

```diff
@@ -53,61 +53,63 @@ def main():
-
-    "$x(i) \in G, G =$"
-    genres
-
-    "The one hot encoding, thus, is something like this"
-    st.table(genre_df.head(10))
-    f"`Actual shape: {genre_df.shape}`"
-
-    """
-        Modifying the original dataframe by including this new genre matrix,
-        (additionally, just joining the `links.csv` data), we have our initial dataset
-    """
-    dataset = pd.concat([df, pd.read_csv("data/links.csv", index_col=0), genre_df], axis=1)
-    del dataset['genres']
-    del dataset['title']
-    st.dataframe(dataset.head(10))
-    cols = list(dataset.columns)
-    st.write(f"This dataset has ${dataset.shape[0]}$ records and {cols} fields")
-
-    st.header("Additional features")
-
-    st.subheader("Average ratings and user count")
-
-    """
-        Now that we have a basic data set with one hot encoded genres, the next step
-        is to add additional information from the `ratings.csv` and optionally
-        `genome-*.csv`.
-    """
-
+    if filename == "data/movies.csv":
+        st.subheader("Year Extraction")
+        """
+            The first step is to include a year by
+            extracting it from the title (_and subsequently remove the year from the title_)
+        """
+
+        df.loc[:, 'name'] = df.title.str.replace(r'\(\d{4}\)', '').str.strip()
+        df.loc[:, 'year'] = df.title.str.extract(r'(\d{4})', expand=False)
+        st.dataframe(df.head(10))
+
+        st.subheader("One Hot Encoding")
+        """
+                        The next step is to create a one-hot encoding for the genres
+                        Consider a function $F(x)$, where $x \in X$, where $X$ is
+                        the set of all possible genre combinations. Now, $F$ is defined as:
+                        $$
+                            F: X \\rightarrow Y,F(x \in X) = y \in Y
+                        $$
+                        where $y$ is defined as a vector such that $y(i) = 1$ if a
+                        genre, x(i) is present for a movie. Thus the elements of $y$
+                        act as a bitmask for the set of genres $G$, such that all $x(i) \in G$
+        """
+
+        import numpy as np
+
+        genre_df, genres = get_onehot(df, return_genres=True)
+
+        "$x(i) \in G, G =$"
+        genres
+
+        "The one hot encoding, thus, is something like this"
+        st.table(genre_df.head(10))
+        f"`Actual shape: {genre_df.shape}`"
+
+        """
+            Modifying the original dataframe by including this new genre matrix,
+            (additionally, just joining the `links.csv` data), we have our initial dataset
+        """
+        dataset = pd.concat([df, pd.read_csv("data/links.csv", index_col=0), genre_df], axis=1)
+        del dataset['genres']
+        del dataset['title']
+        st.dataframe(dataset.head(10))
+        cols = list(dataset.columns)
+        st.write(f"This dataset has ${dataset.shape[0]}$ records and {cols} fields")
+
+
+    elif filename == "data/ratings.csv":
```

Comment:
Just create a new filename parameter for the `ratings.csv` file, something alone [these](https://github.com/Pk13055/recommender-analysis/pull/1/files#diff-3f41e546893dc64b71aaacad12cad815R44-R50) lines